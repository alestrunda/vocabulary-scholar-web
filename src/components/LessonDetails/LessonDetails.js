import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import AnswerPreview from '../AnswerPreview';

const LessonDetails = ({ answers, className, style }) => {
  return (
    <div className={classNames('lesson-details', className)} style={style}>
      <div className={classNames('lesson-details__head', `${className}__head`)}>
        <h3 className="mb10">Lesson details</h3>
      </div>
      {answers.map(answer => (
        <AnswerPreview
          key={answer.word.id}
          word={answer.word.word}
          answered={answer.answered}
        />
      ))}
    </div>
  );
};

LessonDetails.propTypes = {
  answers: PropTypes.array,
  className: PropTypes.string,
  style: PropTypes.object,
};

export default LessonDetails;
