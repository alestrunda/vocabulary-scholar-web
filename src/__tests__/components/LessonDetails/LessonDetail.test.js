import React from 'react';
import { shallow } from 'enzyme';

import LessonDetails from '../../../components/LessonDetails';

it('renders correctly', () => {
  const answers = [
    {
      word: {
        id: 'school',
        word: 'school',
      },
      answered: 'school',
    },
    {
      word: {
        id: 'car',
        word: 'car',
      },
      answered: 'bus',
    },
  ];
  const root = shallow(<LessonDetails answers={answers} />);
  expect(root).toMatchSnapshot();
});

it('renders correctly no items', () => {
  const answers = [];
  const root = shallow(<LessonDetails answers={answers} />);
  expect(root).toMatchSnapshot();
});
