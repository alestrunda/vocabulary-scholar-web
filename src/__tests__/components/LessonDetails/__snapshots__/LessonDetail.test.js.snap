// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LessonDetails
    answers={
      Array [
        Object {
          "answered": "school",
          "word": Object {
            "id": "school",
            "word": "school",
          },
        },
        Object {
          "answered": "bus",
          "word": Object {
            "id": "car",
            "word": "car",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="lesson-details__head undefined__head"
        >
          <h3
            className="mb10"
          >
            Lesson details
          </h3>
        </div>,
        Array [
          <AnswerPreview
            answered="school"
            word="school"
          />,
          <AnswerPreview
            answered="bus"
            word="car"
          />,
        ],
      ],
      "className": "lesson-details",
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3
            className="mb10"
          >
            Lesson details
          </h3>,
          "className": "lesson-details__head undefined__head",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lesson details",
            "className": "mb10",
          },
          "ref": null,
          "rendered": "Lesson details",
          "type": "h3",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "school",
        "nodeType": "function",
        "props": Object {
          "answered": "school",
          "word": "school",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "car",
        "nodeType": "function",
        "props": Object {
          "answered": "bus",
          "word": "car",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="lesson-details__head undefined__head"
          >
            <h3
              className="mb10"
            >
              Lesson details
            </h3>
          </div>,
          Array [
            <AnswerPreview
              answered="school"
              word="school"
            />,
            <AnswerPreview
              answered="bus"
              word="car"
            />,
          ],
        ],
        "className": "lesson-details",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3
              className="mb10"
            >
              Lesson details
            </h3>,
            "className": "lesson-details__head undefined__head",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lesson details",
              "className": "mb10",
            },
            "ref": null,
            "rendered": "Lesson details",
            "type": "h3",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "school",
          "nodeType": "function",
          "props": Object {
            "answered": "school",
            "word": "school",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "car",
          "nodeType": "function",
          "props": Object {
            "answered": "bus",
            "word": "car",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders correctly no items 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LessonDetails
    answers={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="lesson-details__head undefined__head"
        >
          <h3
            className="mb10"
          >
            Lesson details
          </h3>
        </div>,
        Array [],
      ],
      "className": "lesson-details",
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3
            className="mb10"
          >
            Lesson details
          </h3>,
          "className": "lesson-details__head undefined__head",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lesson details",
            "className": "mb10",
          },
          "ref": null,
          "rendered": "Lesson details",
          "type": "h3",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="lesson-details__head undefined__head"
          >
            <h3
              className="mb10"
            >
              Lesson details
            </h3>
          </div>,
          Array [],
        ],
        "className": "lesson-details",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3
              className="mb10"
            >
              Lesson details
            </h3>,
            "className": "lesson-details__head undefined__head",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lesson details",
              "className": "mb10",
            },
            "ref": null,
            "rendered": "Lesson details",
            "type": "h3",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
